### ISSUE TITLE
No MangoHud showing on 32bit windows games when started with linux-runtime soldier

### ISSUE THREAD

**Issue by priand1** [ValveSoftware/steam-runtime#567]
#### Your system information

* Steam Runtime Version: 0.20230118.0
* Distribution (e.g. Ubuntu 18.04): openSUSE Tumbleweed 20230217
* Link to your full system information (Help -> System Information) in a [Gist](https://gist.github.com/): 
[System Infomation](https://gist.github.com/priand1/28eafcb6cb59f30ae982246a54cfa300)
* Have you checked for system updates?: [Yes/No] Yes
* What compatibility tool are you using?: Steam Linux Runtime Soldier / Proton 7.06
* If you are using Steam Linux Runtime for native Linux games: What versions are listed in `SteamLinuxRuntime/VERSIONS.txt`? 0.20230216.0
* If you are using Steam Linux Runtime, or Proton 5.13 or newer: What versions are listed in `SteamLinuxRuntime_soldier/VERSIONS.txt`?  0.20230216.41811

#### Please describe your issue in as much detail as possible:

Expected result:

Mangohud overlay is working with 32bit windows games.

Actual result:

Mangohud overlay is not working with 32bit windows games when started with steam linux-runtime soldier.

The following log was taken using the game 'Floating Point (302380)':

[SLR log no mangohud 32bit with soldier/pv 20230216](https://gist.github.com/priand1/db5b70739832118fe8e19eb4dbd8ccea)

[SLR log no mangohud 32bit with soldier/pv 20230220](https://gist.github.com/priand1/7fd7b86fb9565272ccd98e46e50cb3f1)

#### Steps for reproducing this issue:

1. Just start any 32bit windows game from steam using proton/slr soldier

**Comment by smcv** [comment]
For anyone following along with this, more details are in comments on #564, but I asked @priand1 to open a separate issue, because this didn't look like #564 to me.

**Comment by smcv** [comment]
Did you say that you were also finding that the Steam Overlay didn't work for you? If true, that's a more serious problem, because MangoHud is a nice-to-have for monitoring, but the Steam Overlay is (at least sometimes) functionally necessary for e.g. Steam Input.

**Comment by kisak-valve** [comment]
Reported separately at https://github.com/ValveSoftware/steam-for-linux/issues/9225.

**Comment by smcv** [comment]
@priand1, are you intending to have this bug report represent the missing MangoHud, whereas https://github.com/ValveSoftware/steam-for-linux/issues/9225 represents the missing Steam Overlay?

If yes, then I think I'm going to follow up on https://github.com/ValveSoftware/steam-for-linux/issues/9225 first, because that's the more serious problem; and maybe while trying to solve that, we'll accidentally find out why MangoHud isn't working for you either.

**Comment by smcv** [comment]
In [the first log with p-v 0.20230216.0](https://gist.github.com/priand1/db5b70739832118fe8e19eb4dbd8ccea):

`libMangoApp.so` seems to be missing, for both 64-bit and 32-bit which might indicate a wrong installation:

```
14:52:21.622761: pressure-vessel-wrap[6913]: I: Unable to load library /usr/$LIB/mangohud/libMangoApp.so: Unable to find the library: /usr/$LIB/mangohud/libMangoApp.so: cannot open shared object file: No such file or directory
```

... but that one shouldn't be necessary for the MangoHud overlay.

 I can see that you are affected by #564. We find both 64-bit and 32-bit versions:

```
14:52:21.616714: pressure-vessel-wrap[6913]: I: Evaluating dynamic string tokens in "/usr/$LIB/mangohud/libMangoHud.so"
...
14:52:21.620707: pressure-vessel-wrap[6913]: I: After evaluating dynamic string tokens: "/usr/lib64/mangohud/libMangoHud.so"
```

and

```
14:52:21.717100: pressure-vessel-wrap[6913]: I: Evaluating dynamic string tokens in "/usr/$LIB/mangohud/libMangoHud.so"
...
14:52:21.720488: pressure-vessel-wrap[6913]: I: After evaluating dynamic string tokens: "/usr/lib/mangohud/libMangoHud.so"
```

but then only the 64-bit version is created in the right directory:

```
14:52:21.796897: pressure-vessel-wrap[6913]: D: Generating "overrides/share/vulkan/implicit_layer.d/MangoHud.json" with path "/usr/lib/pressure-vessel/overrides/lib/x86_64-linux-gnu/vulkan_imp_layer/4/libMangoHud.so"
14:52:21.796999: pressure-vessel-wrap[6913]: D: Generating "overrides/share/vulkan/implicit_layer.d/4-i386-linux-gnu/MangoHud.json" with path "/usr/lib/pressure-vessel/overrides/lib/i386-linux-gnu/vulkan_imp_layer/4/libMangoHud.so"
```

Later, we can see that the 64-bit version is loaded:

```
[2023-02-21 14:52:23.308] [MANGOHUD] [info] [config.cpp:118] skipping config: '/home/atze/.config/MangoHud/wine-explorer.conf' [ not found ]
```

but the 32-bit version isn't. That's consistent with #564.

In your [second log with p-v 0.20230220.0](https://gist.github.com/priand1/7fd7b86fb9565272ccd98e46e50cb3f1), we can see that #564 has been fixed:

```
15:21:13.203881: pressure-vessel-wrap[17518]: D: Setting up JSON manifest for "share/vulkan/implicit_layer.d" loadable module "MangoHud.json": "/usr/share/vulkan/implicit_layer.d/MangoHud.json"
15:21:13.203926: pressure-vessel-wrap[17518]: D: Generating "overrides/share/vulkan/implicit_layer.d/4-x86_64-linux-gnu.json" with path "/usr/lib/pressure-vessel/overrides/lib/x86_64-linux-gnu/vulkan_imp_layer/4/libMangoHud.so"
15:21:13.204047: pressure-vessel-wrap[17518]: D: Generating "overrides/share/vulkan/implicit_layer.d/4-i386-linux-gnu.json" with path "/usr/lib/pressure-vessel/overrides/lib/i386-linux-gnu/vulkan_imp_layer/4/libMangoHud.so"
```

Here, both the JSON manifests are in the expected directory, so Vulkan-Loader should pick them up. Their filenames are different now, but that's as intended (we do this intentionally to avoid them overwriting each other).

But then at the end of the log, we can still see 64-bit processes loading MangoHud and the 32-bit game process not, and I don't understand why that is.

**Comment by smcv** [comment]
I think there is a more general 32-bit issue on this system, so I'm going to pause looking into this issue, and I suggest focusing on https://github.com/ValveSoftware/steam-for-linux/issues/9225 instead. After that issue is resolved (or after we understand what's happening there, at least), we can come back to this one.

**Comment by priand1** [comment]
Closed as solved per #564 and [ No Steam overlay with any 32bit game #9225 ](https://github.com/ValveSoftware/steam-for-linux/issues/9225)
